internal/cluster/validations/validations.go

```diff
@@ -208,8 +209,23 @@ func (v *registryPullSecretValidator) ValidatePullSecret(secret string, username
 		}
 	}
 
-	for registry := range *v.registriesWithAuth {
+	registriesWithAuth := make([]string, len(*v.registriesWithAuth))
+	copy(registriesWithAuth, funk.Map(*v.registriesWithAuth, func(registry string, _ bool) string {
+		return registry
+	}).([]string))
 
+	if releaseImageURL != "" {
+		MoreRegistriesWithAuth, err := getRegistriesWithAuth(v.config.PublicRegistries, ignoreListSeparator, releaseImageURL)
+		if err != nil {
+			return err
+		}
+
+		registriesWithAuth = append(registriesWithAuth, funk.Map(*MoreRegistriesWithAuth, func(registry string, _ bool) string {
+			return registry
+		}).([]string)...)
+	}
```

Comment:
Instead of doing this let's extract the logic from `getRegistriesWithAuth` that determines if a particular image's registry needs auth. That function can return (bool, string, error) where the bool indicates if the registry returned in string (parsed from the image) requires auth.

This and making the `v.registriesWithAuth` a slice instead of a map should simplify quite a lot of this.